apps/console/src/features/organization-discovery/components/add-organization-discovery-domains.tsx

```diff
@@ -172,17 +177,59 @@ const AddOrganizationDiscoveryDomains: FunctionComponent<AddOrganizationDiscover
     };
 
     /**
-     * Function to validate the input string is an email domain.
+     * Function to check whether an email domain is available.
+     *
+     * @param values - Email domains.
+     */
+    const checkEmailDomainAvailability = async (emailDomain: string): Promise<boolean> => {
+
+        let available: boolean = true;
+
+        await checkEmailDomainAvailable(emailDomain)
+            .then((response: OrganizationDiscoveryCheckResponseInterface) => {
+                available = response?.available;
+            })
+            .catch(() => {
+                dispatch(
+                    addAlert({
+                        description: t(
+                            "console:manage.features.organizationDiscovery.notifications" +
+                                ".checkEmailDomain.error.description"
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t(
+                            "console:manage.features.organizationDiscovery.notifications" +
+                                ".checkEmailDomain.error.message"
+                        )
+                    })
+                );
+            });
+
+        return available;
+    };
```

Comment:
This function we can rewrite using state variables, and useEffect.

Reply:
Is there any advantage with the suggested approach?